File: 12. 브루트 포스/4. 1018.py
import sys

s = list(map(int, sys.stdin.readline().split()))
row = s[0]
column = s[1]
board = []
for i in range(row):
    board.append(sys.stdin.readline().rstrip())

min = 64
count = 0
for a in range(0, row-7):
    for b in range(0, column-7):
        for case in range(0, 2):
            for i in range(a, a+8):
                for j in range(b, b+8):
                    if case == 0:
                        if i % 2 == 0 and j % 2 == 0:
                            if board[i][j] != "W":
                                count += 1
                        elif i % 2 == 0 and j % 2 == 1:
                            if board[i][j] != "B":
                                count += 1
                        elif i % 2 == 1 and j % 2 == 0:
                            if board[i][j] != "B":
                                count += 1
                        elif i % 2 == 1 and j % 2 == 1:
                            if board[i][j] != "W":
                                count += 1
                    else:
                        if i % 2 == 0 and j % 2 == 0:
                            if board[i][j] != "B":
                                count += 1
                        elif i % 2 == 0 and j % 2 == 1:
                            if board[i][j] != "W":
                                count += 1
                        elif i % 2 == 1 and j % 2 == 0:
                            if board[i][j] != "W":
                                count += 1
                        elif i % 2 == 1 and j % 2 == 1:
                            if board[i][j] != "B":
                                count += 1
            if count < min:
                min = count
            count = 0
print(min)

File: C++/12. 브루트 포스/4. 1018.cpp
#include <iostream>

using namespace std;

int main() {
    int n, m;
    cin >> n >> m;

    string board[50];
    for (int i = 0; i < n; i++) {
        cin >> board[i];
    }
    
    int count, min = 64;
    for (int i = 0; i < n - 7; i++) {
        for (int j = 0; j < m - 7; j++) {
            for (int c = 0; c < 2; c++) {
                count = 0;
                for (int x = i; x < i + 8; x++) {
                    for (int y = j; y < j + 8; y++) {
                        if (c == 0) {
                            if (x % 2 == 0 && y % 2 == 0)
                                if (board[x][y] != 'W')
                                    count++;
                            if (x % 2 == 0 && y % 2 == 1)
                                if (board[x][y] != 'B')
                                    count++;
                            if (x % 2 == 1 && y % 2 == 0)
                                if (board[x][y] != 'B')
                                    count++;
                            if (x % 2 == 1 && y % 2 == 1)
                                if (board[x][y] != 'W')
                                    count++;
                        }
                        else {
                            if (x % 2 == 0 && y % 2 == 0)
                                if (board[x][y] != 'B')
                                    count++;
                            if (x % 2 == 0 && y % 2 == 1)
                                if (board[x][y] != 'W')
                                    count++;
                            if (x % 2 == 1 && y % 2 == 0)
                                if (board[x][y] != 'W')
                                    count++;
                            if (x % 2 == 1 && y % 2 == 1)
                                if (board[x][y] != 'B')
                                    count++;
                        }
                    }
                }
                if (count < min)
                    min = count;
            }
        }
    }

    cout << min << endl;
    
    return 0;
}